app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,110 @@
+         * @param performanceInfo Whether to return decode time taken or not
+         */
+        Result DecodeInterleavedImpl(ipc::IpcRequest &request, ipc::IpcResponse &response, bool performanceInfo = false);
+
+      public:
+        /**
+         * @param sampleRate The sample rate of the Opus audio data
+         * @param channelCount The channel count of the Opus audio data
+         */
+        IHardwareOpusDecoder(const DeviceState &state, ServiceManager &manager, i32 sampleRate, i32 channelCount, u32 workBufferSize, KHandle kWorkBuffer);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size and decoded sample count
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedOld
+         */
+        Result DecodeInterleavedOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleavedWithPerfOld
+         */
+        Result DecodeInterleavedWithPerfOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Same as DecodeInterleaved
+         * @see DecodeInterleaved
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleavedWithPerfAndResetOld(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        /**
+         * @brief Takes an input boolean flag, an OpusDataIn input buffer and a PcmDataOut output buffer.
+         * Decodes the Opus source data to the output buffer and returns decoded data size, decoded sample count and decode time taken in microseconds
+         * @note The bool flag indicates whether or not a reset of the decoder context is being requested
+         * @url https://switchbrew.org/wiki/Audio_services#DecodeInterleaved
+         */
+        Result DecodeInterleaved(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+      SERVICE_DECL(
+          SFUNC(0x0, IHardwareOpusDecoder, DecodeInterleavedOld),
+          SFUNC(0x4, IHardwareOpusDecoder, DecodeInterleavedWithPerfOld),
+          SFUNC(0x6, IHardwareOpusDecoder, DecodeInterleavedWithPerfAndResetOld),
+          SFUNC(0x8, IHardwareOpusDecoder, DecodeInterleaved),
+      )
+    };
+
+  class OpusException : public skyline::exception {
+    public:
+      OpusException(const std::string& message) : skyline::exception(message) {}
+      OpusException(int errorCode) : skyline::exception("Opus failed with error code {}: {}", errorCode, opus_strerror(errorCode)) {}
+  };
```

Comment:
A comment would be nice `@brief A wrapper around libopus error codes for exceptions` and remove the `std::string` constructor since this should only ever be used with libopus error codes